tests/support/util.tcl

```diff
@@ -666,6 +666,8 @@ proc generate_fuzzy_traffic_on_key {key duration} {
     set commands [dict create string $string_commands hash $hash_commands zset $zset_commands list $list_commands set $set_commands stream $stream_commands]
 
     set type [r type $key]
+    set dump [r dump $key]
+    set printable_dump [string2printable $dump]
```

Comment:
i don't like this (re-generate the dump),
it works nicely for encoded keys (listpack), but will not work in case we decode or convert the key.

i think that what you could do is re-throw the SIGTERM and catch it again in the caller where we have the original dump.

the other alternative is to pass some "context" argument to this function to be printed in that case.

Reply:
generate_fuzzy_traffic_on_key acts on the $key, so the dump is the target (origin) one
i think with this payload and the sent commands list, we would be able to re-run it

"re-generate the dump", do you mean it call dump twice or call string2printable twice is the problem?

Reply:
imagine the RESTORE payload in corrupt-dump-fuzzser would send a listpack encoded object, but restore would for some reason decide to convert it to dict encoding, and then you run DUMP on it, you'll get a different payload than the one used for RESTORE.
i want to print the actual RESTORE command we run, and not use DUMP to get it.

Reply:
i had a few minutes (and i guess i miss coding.. not that tcl tests are really codoing), took the liberty to patch your PR, please take a look.

Reply:
got it. This lgtm, thanks for the patch, please feel free to do anything that can make the code better. 

If this not in a hurry to merge, I will try to see its printing effect later. coding is great! It might be hard to figure out the reason at first time, but it is great when we catch it

Reply:
the hang thing output (look great):
```
Testing integration/corrupt-dump-fuzzer
[TIMEOUT]: clients state report follows.
sock6 => (SPAWNED SERVER) pid:28884
Killing still running Redis server 28884
commands caused test to hang:
SDIFF __key 
payload that caused test to hang: "\x14\x25\x25\x00\x00\x00\x0A\x00\x06\x01\x82\x5F\x35\x03\x04\x01\x82\x5F\x31\x03\x82\x5F\x33\x03\x00\x01\x82\x5F\x82\x03\x82\x5F\x33\x03\x08\x01\x02\x01\xFF\x0B\x00\x31\xBE\x7D\x41\x01\x03\x5B\xEC"
```

Reply:
i tested it by passing `--timeout 1` (and commenting out the timeout sanity check in corrupt-dump-fuzzer.tcl)